modules/core/shared/src/main/scala-3/io/circe/derivation/ConfiguredDecoder.scala

```diff
@@ -70,7 +71,11 @@ trait ConfiguredDecoder[A](using conf: Configuration) extends Decoder[A]:
                     s"expected a single key json object with one of: ${constructorNames.iterator.mkString(", ")}."
                   )
                 )
-              else fromName(sumTypeName, c.downField(sumTypeName))
+              else
+                val cursor = c.downField(sumTypeName)
+                constructorNames.indexOf(sumTypeName) match
+                  case -1    => nameNotFound(sumTypeName, cursor)
+                  case index => decode(elemDecoders(index).asInstanceOf[Decoder[A]])(cursor)
```

Comment:
FYI: this is the implementation of `def fromName` from before https://github.com/circe/circe/pull/2098/files#diff-378e58dd77c3ec12a0e0b230ec5d55a15679e08f37f01ef83c5ee06967ca42d4 (Since that change only changed the encoding for the case where `conf.discrimniator.isDefined`, this restores the previous decoding for the `conf.discrimniator.isEmpty` case.)